### ISSUE TITLE
Not connecting to cPanel server?

### ISSUE THREAD

**Issue by samgreco** [WHSuite/whsuite#6]
I posted this in the cPanel add-on area, but got no response, so I thought I would try here.

I have installed WHSuite on my dedicated hosting server running WHM/cPanel.  I have the Remote Access Key in place and everything seems correct.  But it just keeps spinning the "loading" icon.

Manage Server just shows the server host name.  If I go to manage products, it just spins loading continually.

If anyone has any guidance I would truly appreciate it.

**Comment by bigalgeorge** [comment]
Yes, I agree. Its odd, Im going to check out the php code soon see if I can find where its bugging. Whats happening here is when I set up SERVER these are the parameters that work:
`Name: 	Elcheapo Cpanel
Hostname: 	localhost
Main IP: 	148.163.xxx.x //WHM IP adds
Location: 	2086  //port
Username: 	myusername `
and password and key from WHM. Then Manage Server tab shows its connected by displaying the correct server (if you dont get a server, its not working).
However, when entering a new product either using prefix_PRODUCT or PRODUCT, when Hosting Details tab is clicked the progress cog never stops. This shows there is an error communicating product with WHM/Cpanel but like you I cant get to the bottom of it (yet).

**Comment by rickmills** [comment]
@samgreco what fields did you fill out on the server details page? We never got around to tidying it up as some of the fields arent needed for cPanel, and some are. Although if you got past that page, it would indicate that it connected to cPanel/WHM ok.

**Comment by samgreco** [comment]
Unfortunately, I needed to get something up and running and wasn't getting any response here, so I bought into Blesta.  I may revisit this at some point.  The software looks really great.  

But I currently only have the one WHM server.

**Comment by bigalgeorge** [comment]
samgreco you could have used thehostingtool reworked which works well and is free. Whsuite shows more potential though.
Hi Rick here are my WHM server settings (plush a hash tag):
Name: 	Elcheapo Cpanel
Hostname: 	elcheapohost.com
Main IP: 	148.163.124.3
Location: 	:2086/
The above connects and Im not sure if location should be a port or not. Even though it connects to server the problem occurs when ordering or anything that needs the WHM interface.

**Comment by smygdanis** [comment]
I have the same problem, the suite connects to whm but when i create a product it's stuck on "loading". Any news

**Comment by ctmattice** [comment]
For those having problems with this, I have not found an answer but it does work correctly with the following scenario.

1) the server you are connecting to is a VPS or dedicated server.
2) when you setup whsuite you must use a domain name that contains the VPS or dedicated servers primary domain. (ie your server hostname is "this.server.example" and the site url to use with whsuite install is "server.example"
3) when you setup a server you have to use "root" for the username.
4) you must use the servers access key (not a password enter).

If you don't have access to roots access key your out of luck. Best I can determine at this stage of the game (and this is only a guess) is that when trying to access the  requested server to setup a product that whsuite is requesting the info from cPanel but it never gets to your server. Reason why I think is that whsuite is using domain names (or possibly the site name you setup for whsuite) instead of IP addresses. to access the requested server.
